shim/integration/kvm_set_irqchip.cpp

```diff
@@ -0,0 +1,70 @@
+/// @copyright
+/// Copyright (C) 2020 Assured Information Security, Inc.
+///
+/// @copyright
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// @copyright
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+///
+/// @copyright
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+/// SOFTWARE.
+
+#include <integration_utils.hpp>
+#include <ioctl_t.hpp>
+#include <shim_platform_interface.hpp>
+
+#include <bsl/convert.hpp>
+#include <bsl/enable_color.hpp>
+#include <bsl/exit_code.hpp>
+#include <bsl/safe_idx.hpp>
+#include <bsl/safe_integral.hpp>
+
+/// <!-- description -->
+///   @brief Provides the main entry point for this application.
+///
+/// <!-- inputs/outputs -->
+///   @return bsl::exit_success on success, bsl::exit_failure otherwise.
+///
+
+[[nodiscard]] auto
+main() noexcept -> bsl::exit_code
+{
+    //@ brief defining VAL for just sending an argument to the write function
+    constexpr auto val{42_u64};
+    bsl::safe_i64 mut_ret{};
+    bsl::enable_color();
+    integration::ioctl_t mut_system_ctl{shim::DEVICE_NAME};
+    /// Verify that get/set works
```

Comment:
Please use the double forward slash comments when commenting within the body of a function. Triple forward slash style comments are for documentation.